### PR TITLE
Switch DisruptionBudget api from bool to int allowed disruptions [only v1beta1]

### PR DESCRIPTION
**Pull request by mwielgus**
Continuation of #34546. Apparently it there is some bug that prevents us from having 2 different incompatibile version of API in integration tests. So in this PR v1alpha1 is removed until testing infrastructure is fixed.

Base PR comment:

Currently there is a single bool in disruption budget api that denotes whether 1 pod can be deleted or not. Every time a pod is deleted the apiserver filps the bool to false and the disruptionbudget controller sets it to true if more deletions are allowed. This works but it is far from optimal when the user wants to delete multiple pods (for example, by decreasing replicaset size from 10000 to 8000).
This PR adds a new api version v1beta1 and changes bool to int which contains a number of pods that can be deleted at once.

cc: @davidopp @mml @wojtek-t @fgrzadkowski @caesarxuchao

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35567)

### CODE REVIEW

**Comment by mwielgus** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by wojtek-t** [comment]
Please fix ```verify```.

File: pkg/registry/core/pod/etcd/eviction.go [hunk 1]
```diff
@@ -103,11 +103,10 @@ func (r *EvictionREST) Create(ctx api.Context, obj runtime.Object) (runtime.Obje
 }
 
 func (r *EvictionREST) checkAndDecrement(namespace string, pdb policy.PodDisruptionBudget) (ok bool, err error) {
-	if !pdb.Status.PodDisruptionAllowed {
+	if pdb.Status.PodDisruptionsAllowed <= 0 {
```
Comment (wojtek-t) [inline pkg/registry/core/pod/etcd/eviction.go]: We should return error if this is smaller than 0.

File: pkg/apis/policy/v1beta1/doc.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
+/*
```
Comment (wojtek-t) [inline pkg/apis/policy/v1beta1/doc.go]: Please do NOT do this in future - this commit is super huge and it contains both autogenerated code as well as manually writte - this is unreviable.

Comment (mwielgus) [inline pkg/apis/policy/v1beta1/doc.go]: Must have been included by accident.

**Comment by mwielgus** [comment]
@k8s-bot node e2e test this

**Comment by mwielgus** [comment]
@k8s-bot node e2e test this issue: #36159

**Pull request merged**